src/datepicker/day.js

```diff
@@ -83,39 +84,64 @@ export default class Day<T = Date> extends React.Component<
       : this.props.month;
   };
 
```

Comment:
The logic for determining what happens when a user clicks on a day in the calendar is spread out across three components:`Day`, `Calendar`, and `Datepicker`. 

`Day`'s `onSelect` method determines whether a selected day is considered a Start Date or End Date.
`Calendar`'s `handleDateChange` merges time values in with the day values to produce the full date.
`Datepicker`'s `onChange` method handles edge cases like Start Dates that fall after End Dates, and also manages whether the calendar should stay open or not after a selection.

It might make sense to organize this logic in a different way in the future, but for now I've tried to conform to the existing rough separation of concerns.